vividus/src/main/java/org/vividus/runner/ScenariosCounter.java

```diff
@@ -0,0 +1,116 @@
+import org.vividus.configuration.BeanFactory;
+import org.vividus.configuration.Vividus;
+
+// Fix once deprecated child class is removed
+@SuppressWarnings({ "checkstyle:HideUtilityClassConstructor", "PMD.UseUtilityClass" })
+public class ScenariosCounter
+{
+    private static final String DEFAULT_STORY_LOCATION = "story";
+    private static final String FORMATTER = "%5d | %s";
+
+    public static void main(String[] args) throws ParseException, InitializationError, ReflectiveOperationException
+    {
+        Vividus.init();
+        CommandLineParser parser = new DefaultParser();
+        Option helpOption = new Option("h", "help", false, "print this message.");
+        Option directoryOption = new Option("d", "dir", true, "directory to count scenarios in (e.g. story/release).");
+        Options options = new Options();
+        options.addOption(helpOption);
+        options.addOption(directoryOption);
+        CommandLine commandLine = parser.parse(options, args);
+        if (commandLine.hasOption(helpOption.getOpt()))
+        {
+            new HelpFormatter().printHelp("ScenariosCounter", options);
+            return;
+        }
+
+        String storyLocation = commandLine.hasOption(directoryOption.getOpt())
+                ? commandLine.getOptionValue(directoryOption.getOpt()) : DEFAULT_STORY_LOCATION;
+        configureStoryLocation(storyLocation);
+
+        System.out.println("Story parsing may take up to 5 minutes. Please be patient.");
+        JUnit4StoryRunner runner = new JUnit4StoryRunner(StoriesRunner.class);
+
+        print(getNumberOfChildren(runner.getDescription(), Level.STORY.getLevel()), "Stories");
+        print(getNumberOfChildren(runner.getDescription(), Level.SCENARIO.getLevel()), "Scenarios");
+        print(getNumberOfChildren(runner.getDescription(), Level.EXAMPLE.getLevel()), "Scenarios with Examples");
+    }
+
+    private static int getNumberOfChildren(Description description, int level)
+    {
+        if (level < 1)
+        {
+            return 1;
+        }
+        int childrenNumber = 0;
+        for (Description childDescription : description.getChildren())
+        {
+            if (childDescription.getMethodName() == null)
+            {
+                childrenNumber += getNumberOfChildren(childDescription, level - 1);
+            }
+        }
+        return childrenNumber > 0 ? childrenNumber : 1;
+    }
+
+    private static void print(int number, String message)
+    {
+        System.out.printf(FORMATTER, number, message);
+        System.out.println();
+    }
+
+    private static void configureStoryLocation(String storyLocation)
+    {
+        Properties properties = BeanFactory.getBean("properties", Properties.class);
+        properties.put("bdd.story-loader.batch1.resource-location", storyLocation);
+        properties.put("bdd.story-loader.batch1.resource-include-patterns", "**/*.story");
+        properties.put("bdd.story-loader.batch1.resource-exclude-patterns", "");
```

Comment:
should not it be changed as well?

Reply:
nope, not in this PR, the goal of this PR is to update naming of classes, and then switch to use new classes in build system 2.0